arcusApp/app/src/main/java/arcus/app/device/adapter/DeviceDetailPagerAdapter.java

```diff
@@ -184,6 +184,8 @@ private Fragment getFragmentForDevice(@NonNull DeviceModel device) {
             case SMOKE:
                 product = SmokeDetectorFragment.newInstance(false);
                 break;
+            case THERMOSTAT:
+                break;
```

Comment:
Is this intentional? Shouldn't this be left to the default block so that `product` isn't left unassigned? Otherwise, this will throw an NPE if a device with a `THERMOSTAT` `DeviceType` is detected when it tries to set the device `product.setDeviceModel(device);` (and after).

Reply:
I'm actually getting a compile error using this since `product` is not initialized so this should be removed or `product` needs to be assigned to something in this block.

Reply:
product now initialized to ThermostatFragment